src/main/java/seedu/pill/util/ItemMap.java

```diff
@@ -1,21 +1,24 @@
 package seedu.pill.util;
 
+import java.time.LocalDate;
+import java.util.Optional;
 import java.util.LinkedHashMap;
 import java.util.Iterator;
 import java.util.Map;
+import java.util.TreeSet;
 import java.util.logging.Logger;
 import java.util.logging.Level;
 
 /**
  * Represents a list of items and provides methods to add, delete, list, and edit items.
  */
-public class ItemMap implements Iterable<Map.Entry<String, Item>> {
+public class ItemMap implements Iterable<Map.Entry<String, TreeSet<Item>>> {
     private static final Logger LOGGER = Logger.getLogger(ItemMap.class.getName());
-    private Map<String, Item> items;
+    private Map<String, TreeSet<Item>> items;
 
     /**
-     * Constructor for ItemList.
-     * Initializes the internal ArrayList to store items.
+     * Constructor for ItemMap.
+     * Initializes the internal Map to store items.
```

Comment:
Thanks for correcting the JavaDocs. I missed out on these in my previous commits.